video_xblock/static/js/studio_edit.js

```diff
@@ -157,37 +157,141 @@ function StudioEditableXBlock(runtime, element) {
+                field.removeEditor();
+            }
+        }
+        e.preventDefault();
+        runtime.notify('cancel', {});
+    });
+
     // Raccoongang addons
 
+    var languages = [];
     var $fileUploader = $('.input-file-uploader', element);
+    var $transcriptUploader = $('.input-transcript-uploader', element);
+    var $langChoicer = $('.lang-choiser', element);
+    var setLanguages = $('input[data-field-name="transcripts"]').val();
+
+    if(setLanguages){
+        languages = JSON.parse(setLanguages);
+    }
+    var pushLanguage = function (lang, label, url){
+        var langExists = false;
+        for (var i=0; i < languages.length; i++){
+            if (lang == languages[i].lang){
+                langExists = true
+            }
+        }
+        if (!langExists){
+            languages.push({
+                lang: lang,
+                url: url,
+                label: label
+            })
+        }
+    };
 
     $fileUploader.on('change', function(e) {
-        var fieldName = $fileUploader.data('change-field-name');
-        $('form', element).ajaxSubmit({
+        var fieldName = $(e.currentTarget).data('change-field-name');
+        $('.file-uploader-form', element).ajaxSubmit({
             success: function(response, statusText, xhr, form) {
                 $('input[data-field-name=' + fieldName + ']').val(response['asset']['id']).change();
             }
+        });
+    });
+
+    $transcriptUploader.on('change', function (event) {
+        var lang = $(event.currentTarget).data('lang-code');
+        var label = $(event.currentTarget).data('lang-label');
+        $('.transcript-uploader-form', element).ajaxSubmit({
+            success: function(response, statusText, xhr, form) {
+                pushLanguage(lang, label, '/' + response['asset']['id']);
+                $('input[data-field-name="transcripts"]').val(JSON.stringify(languages)).change();
+            }
         })
     });
 
-    $('.upload-setting', element).on('click', function(e) {
+    $('.upload-action', element).on('click', function(e) {
         e.preventDefault();
         var fieldName = $(e.currentTarget).data('change-field-name');
-        $fileUploader.attr('data-change-field-name', fieldName);
+        $fileUploader.data('change-field-name', fieldName);
         $fileUploader.click();
     });
-    // End of Raccoongang addons
 
-    $(element).find('.cancel-button').bind('click', function(e) {
-        // Remove TinyMCE instances to make sure jQuery does not try to access stale instances
-        // when loading editor for another block:
-        for (var i in fields) {
-            var field = fields[i];
-            if (field.hasEditor()) {
-                field.removeEditor();
+    $('.add-transcript', element).on('click', function (event) {
+        event.preventDefault();
+
+        var $choiserItem = $('.list-settings-item:hidden').clone();
+        $choiserItem.removeClass('hidden');
```

Comment:
you should use `is-hidden` instead of `hidden`. See https://smacss.com/book/type-state